src/dash-table/handlers/cellEvents.ts

```diff
@@ -1,57 +1,69 @@
-import * as R from 'ramda';
-import { SelectedCells, ICellFactoryProps } from 'dash-table/components/Table/props';
+import { min, max, set, lensPath } from 'ramda';
+import { ICellFactoryProps } from 'dash-table/components/Table/props';
 import isActive from 'dash-table/derived/cell/isActive';
+import isSelected from 'dash-table/derived/cell/isSelected';
+import { makeCell, makeSelection } from 'dash-table/derived/cell/cellProps';
 import reconcile from 'dash-table/type/reconcile';
 
-function isCellSelected(selectedCells: SelectedCells, idx: number, i: number) {
-    return selectedCells && R.contains([idx, i], selectedCells);
-}
-
 export const handleClick = (propsFn: () => ICellFactoryProps, idx: number, i: number, e: any) => {
     const {
         selected_cells,
+        start_cell,
         setProps,
-        virtualized
+        virtualized,
+        columns,
+        viewport
     } = propsFn();
 
-    const selected = isCellSelected(selected_cells, idx, i);
+    const selected = isSelected(selected_cells, idx, i);
 
     // don't update if already selected
     if (selected) {
         return;
     }
 
     e.preventDefault();
-    const cellLocation: [number, number] = [
-        idx + virtualized.offset.rows,
-        i + virtualized.offset.columns
-    ];
+
+    const row = idx + virtualized.offset.rows;
+    const col = i + virtualized.offset.columns;
+
+    const activeCell = makeCell(row, col, columns, viewport);
 
     const newProps: Partial<ICellFactoryProps> = {
         is_focused: false,
-        active_cell: cellLocation
+        active_cell: activeCell,
+        start_cell: activeCell,
+        end_cell: activeCell
     };
 
-    const selectedRows = R.uniq(R.pluck(0, selected_cells)).sort((a, b) => a - b);
-    const selectedCols = R.uniq(R.pluck(1, selected_cells)).sort((a, b) => a - b);
-    const minRow = selectedRows[0];
-    const minCol = selectedCols[0];
-
-    if (e.shiftKey) {
-        newProps.selected_cells = R.xprod(
-            R.range(
-                R.min(minRow, cellLocation[0]),
-                R.max(minRow, cellLocation[0]) + 1
-            ),
-            R.range(
-                R.min(minCol, cellLocation[1]),
-                R.max(minCol, cellLocation[1]) + 1
-            )
-        ) as any;
+    if (e.shiftKey && start_cell && start_cell.row !== undefined) {
+        newProps.selected_cells = makeSelection(
+            {
+                minRow: min(row, start_cell.row),
+                maxRow: max(row, start_cell.row),
+                minCol: min(col, start_cell.column),
+                maxCol: max(col, start_cell.column)
```

Comment:
Oh hmm, here I used `start_cell` and not `active_cell` like I did in the keypress handler... and indeed it can have weird effects like `active_cell` becoming outside the selection. So one of these should change... which one?

Reply:
standardized on `active_cell` like keypresses https://github.com/plotly/dash-table/pull/412#discussion_r277207194 -> https://github.com/plotly/dash-table/pull/412/commits/1c69cfbe8080524946f53d720ae5535886666136